packages/core/src/controllers/publisher.ts

```diff
@@ -39,7 +41,17 @@ export class Publisher extends IPublisher {
       const params = { topic, message, opts: { ttl, relay, prompt, tag } };
       const hash = hashMessage(message);
       this.queue.set(hash, params);
-      await this.rpcPublish(topic, message, ttl, relay, prompt, tag);
+      try {
+        const publish = await createExpiringPromise(
+          this.rpcPublish(topic, message, ttl, relay, prompt, tag),
+          this.publishTimeout,
+        );
+        await publish;
+      } catch (err) {
+        this.logger.debug(`Publishing Payload stalled`);
+        this.relayer.events.emit(RELAYER_EVENTS.connection_stalled);
```

Comment:
do you need to inspect the error here to tell for sure it's the `connection_stalled` one?

Reply:
This event is emitted when the `publish` request fails to resolve within the expiry (currently 10s). In normal circumstances, the expiry should be more than enough for a request to complete. If it doesn't, it is safe to assume the connection stalled. Also, the error here comes from the expiry rejecting the promise, not an actual network error